### PR TITLE
add segment rules

### PR DESCRIPTION
**Pull request by hiaashna**

### CODE REVIEW

File: rules/sinks/third_parties/sdk/segment/java.yaml [hunk 1]
```diff
@@ -4,10 +4,74 @@
 
 sinks:
 
-  - id: ThirdParties.SDK.Segment
-    name: Segment
+  - id: ThirdParties.SDK.Segment.Android
```
Comment (khemrajrathore) [inline rules/sinks/third_parties/sdk/segment/java.yaml:7]: remove this pattern `com[.]segment[.]analytics[.]java`

File: rules/sinks/third_parties/sdk/segment/java.yaml [hunk 2]
```diff
@@ -4,10 +4,74 @@
...
+    patterns:
+      - "(?i)(com[.]segment[.]analytics[.]java).*"
+    tags:
+
+  - id: ThirdParties.SDK.Segment.Javascript
+    name: Segment Analytics Javascript
+    domains:
+      - "segment.com"
+    patterns:
+      - "(?i)(com[.]segment[.]analytics[.]javascript).*"
```
Comment (khemrajrathore) [inline rules/sinks/third_parties/sdk/segment/java.yaml:28]: Is this a Java Package?? 

As this rule file is `java.yaml` only java rule's should be present here

Similar comment for below rule for `python, ruby` etc

File: rules/sinks/third_parties/sdk/segment/java.yaml (hunk 1 above)
Comment (hiaashna) [reply rules/sinks/third_parties/sdk/segment/java.yaml:7]: removed

File: rules/sinks/third_parties/sdk/segment/java.yaml (hunk 2 above)
Comment (hiaashna) [reply rules/sinks/third_parties/sdk/segment/java.yaml]: removed non java rules

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Comment by khemrajrathore** [comment]
@hiaashna browse little more, do we need to add more segment sdk's

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**